Rasberry with Ubuntu Mate 18.04.2 64bits

Hello,

I have installed all the required library for compile the source of planetary imager v0.7.0.
The cmake files were well generated but when I tried to perform the build of the soft by "make" command, this last one crashes at 71% on the build of planetaryimager_mainwindow.cpp.o.

How can I solve this problem?

Thanks

I have the following issue:

[ 64%] Building CXX object src/CMakeFiles/planetary_imager.dir/planetaryimager_mainwindow.cpp.o
c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
src/CMakeFiles/planetary_imager.dir/build.make:86: recipe for target 'src/CMakeFiles/planetary_imager.dir/planetaryimager_mainwindow.cpp.o' failed
make[2]: *** [src/CMakeFiles/planetary_imager.dir/planetaryimager_mainwindow.cpp.o] Error 4
CMakeFiles/Makefile2:345: recipe for target 'src/CMakeFiles/planetary_imager.dir/all' failed
make[1]: *** [src/CMakeFiles/planetary_imager.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2
astrojul@astrojul-desktop:~/build$

According to your advice on power issue, I have removed the power of 1A on the USB port. The file planetaryimager_mainwindow.cpp.o has now well compiled but I have the following issue on the ASI driver:

[ 89%] Linking CXX shared module libzwo_asi_imager.so
/usr/bin/ld: /home/astrojul/PlanetaryImager/src/drivers/zwo_asi/ASI_LINUX_MAC_SDK_v0.6.0921//lib/x64/libASICamera2.a(ASICamera2.o): Relocations in generic ELF (EM: 62)
/usr/bin/ld: /home/astrojul/PlanetaryImager/src/drivers/zwo_asi/ASI_LINUX_MAC_SDK_v0.6.0921//lib/x64/libASICamera2.a(ASICamera2.o): Relocations in generic ELF (EM: 62)
/usr/bin/ld: /home/astrojul/PlanetaryImager/src/drivers/zwo_asi/ASI_LINUX_MAC_SDK_v0.6.0921//lib/x64/libASICamera2.a(ASICamera2.o): Relocations in generic ELF (EM: 62)
/home/astrojul/PlanetaryImager/src/drivers/zwo_asi/ASI_LINUX_MAC_SDK_v0.6.0921//lib/x64/libASICamera2.a: error adding symbols: File in wrong format
collect2: error: ld returned 1 exit status
src/drivers/zwo_asi/CMakeFiles/zwo_asi_imager.dir/build.make:273: recipe for target 'src/drivers/zwo_asi/libzwo_asi_imager.so' failed
make[2]: *** [src/drivers/zwo_asi/libzwo_asi_imager.so] Error 1
CMakeFiles/Makefile2:738: recipe for target 'src/drivers/zwo_asi/CMakeFiles/zwo_asi_imager.dir/all' failed
make[1]: *** [src/drivers/zwo_asi/CMakeFiles/zwo_asi_imager.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2

--- a/cmake/utils.cmake
+++ b/cmake/utils.cmake
@@ -3,6 +3,8 @@
   set(PlanetaryImager_ARCH armv6)
  elseif(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "armv7")
   set(PlanetaryImager_ARCH armv7)
+ elseif(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "aarch64|armv8")
+  set(PlanetaryImager_ARCH armv8)
  elseif(CMAKE_SIZEOF_VOID_P EQUAL "8")
   set(PlanetaryImager_ARCH x86_64)
  else()

--- a/src/drivers/zwo_asi/CMakeLists.txt
+++ b/src/drivers/zwo_asi/CMakeLists.txt
@@ -26,6 +26,7 @@ endif()
 add_definitions(-DASI_SDK_VERSION="${ASI_SDK_VERSION}")
 set(ASI_ARCH_x86_64 x64)
 set(ASI_ARCH_i686 x86)
+set(ASI_ARCH_armv8 armv8)
 set(ASI_ARCH_armv7 armv7)
 set(ASI_ARCH_armv6 armv6)
 

